src/compiler/checker.ts

```diff
@@ -17535,11 +17535,10 @@ namespace ts {
         }
 
         function checkSourceFile(node: SourceFile) {
-            const start = performance.mark();
-
+            performance.mark("checkStart");
             checkSourceFileWorker(node);
-
-            performance.measure("Check", start);
+            performance.mark("checkEnd");

```

Comment:
question: it seems you don't have to call mark here then as the function 'measure' will call `timestamp()` if you don't pass the end mark anyway?